crypto/pkcs12/p12_mutl.c

```diff
@@ -178,7 +291,32 @@ static int pkcs12_gen_mac(PKCS12 *p12, const char *pass, int passlen,
 int PKCS12_gen_mac(PKCS12 *p12, const char *pass, int passlen,
                    unsigned char *mac, unsigned int *maclen)
 {
-    return pkcs12_gen_mac(p12, pass, passlen, mac, maclen, NULL);
+    return pkcs12_gen_mac(p12, pass, passlen, mac, maclen, NID_undef, NID_undef, NULL);
+}
+
+static int pkcs12_hmacnid2mdnid(int hmac_nid)
+{
+    int md_nid = NID_undef;
+
+    switch(hmac_nid) {
+        case NID_hmacWithSHA1:                    md_nid = NID_sha1;                  break;
+        case NID_hmacWithMD5:                     md_nid = NID_md5;                   break;
+        case NID_hmacWithSHA224:                  md_nid = NID_sha224;                break;
+        case NID_hmacWithSHA256:                  md_nid = NID_sha256;                break;
+        case NID_hmacWithSHA384:                  md_nid = NID_sha384;                break;
+        case NID_hmacWithSHA512:                  md_nid = NID_sha512;                break;
+        case NID_id_HMACGostR3411_94:             md_nid = NID_id_GostR3411_94;       break;
+        case NID_id_tc26_hmac_gost_3411_2012_256: md_nid = NID_id_GostR3411_2012_256; break;
+        case NID_id_tc26_hmac_gost_3411_2012_512: md_nid = NID_id_GostR3411_2012_512; break;
+        case NID_hmac_sha3_224:                   md_nid = NID_sha3_224;              break;
+        case NID_hmac_sha3_256:                   md_nid = NID_sha3_256;              break;
+        case NID_hmac_sha3_384:                   md_nid = NID_sha3_384;              break;
+        case NID_hmac_sha3_512:                   md_nid = NID_sha3_512;              break;
+        case NID_hmacWithSHA512_224:              md_nid = NID_sha512_224;            break;
+        case NID_hmacWithSHA512_256:              md_nid = NID_sha512_256;            break;
```

Comment:
I am not sure I like the formatting. Although it makes the _table_ in the switch more concise it is strictly speaking against our code style.

Reply:
Fixed